Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -73,140 +73,184 @@ class WalletConnectManager {
                 )
             } catch {
                 print("DAPP: Respond Error: \(error.localizedDescription)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Respond Error: \(error.localizedDescription)")
+                }
             }
         }
     }
-
+    
     func approveSession(proposal: Session.Proposal) {
         Task {
             guard let safe = try? Safe.getSelected() else { return }
+            
             var sessionNamespaces = [String: SessionNamespace]()
             proposal.requiredNamespaces.forEach {
                 let caip2Namespace = $0.key
                 let proposalNamespace = $0.value
-                let chains = proposalNamespace.chains
-
-                let accounts = Set(chains.compactMap {
+                guard let chains = proposalNamespace.chains else { return }
+                
+                let selectedSafeChain = chains.filter { chain in
+                    chain.namespace == "eip155" && chain.reference == safe.chain?.id
+                }
+                
+                let accounts = Set(selectedSafeChain.compactMap {
                     Account($0.absoluteString + ":\(safe.addressValue)")
                 })
-
+                
                 let sessionNamespace = SessionNamespace(accounts: accounts,
                                                         methods: proposalNamespace.methods,
                                                         events: proposalNamespace.events)
                 sessionNamespaces[caip2Namespace] = sessionNamespace
             }
             do {
-                try await Sign.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
+                try await Web3Wallet.instance.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
             } catch {
                 print("DAPP: Approve Session error: \(error)")
+                Task { @MainActor in
+                    App.shared.snackbar.show(message: "DAPP: Approve Session error: \(error)")
+                }
             }
         }
     }
-
+    
     /// By default, session lifetime is set for 7 days and after that time user's session will expire.
     /// This method will extend the session for 7 days
     func extend(session: Session) async {
         do {
             try await Web3Wallet.instance.extend(topic: session.topic)
         } catch {
+            Task { @MainActor in
+                App.shared.snackbar.show(message: error.localizedDescription)
+            }
             print("DAPP: extending Session error: \(error)")
         }
     }
-
+    
     func disconnect(session: Session) {
         Task {
             do {
                 NotificationCenter.default.post(name: .wcDidDisconnectSafeServer, object: self)
-                try await Sign.instance.disconnect(topic: session.topic)
+                try await Web3Wallet.instance.disconnect(topic: session.topic)
```

Comment:
Maybe if we disconnect pairing instead of disconnect session it might solve the problem ?, not sure about that

Reply:
All unused pairings are disconnected in `disconnectUnusedPairings()` below.